src/ServerService.php

```diff
@@ -0,0 +1,140 @@
+        if (empty($serverIdentifier)) {
+            $serverIdentifier = self::SERVER_SERVICE_PREFIX . $identifier;
+        }
+        parent::__construct($config, $serverIdentifier, $types, $description);
+    }
+
+    /**
+     * Returns the class name of the client service.
+     *
+     * @return string The class name of the client service.
+     */
+    public function getClientClassName()
+    {
+        return $this->clientClassName;
+    }
+
+    /**
+     * Sets the class name of the client service.
+     *
+     * @return string The class name of the client service.
+     */
+    public function setClientClassName($clientClassName)
+    {
+        $this->clientClassName = $clientClassName;
+        $this->class = null;
+    }
+
+    /**
+     * Generates the class if not already generated
+     */
+    public function generateClass()
+    {
+        $serverName = $this->identifier;
+
+        // Generate a valid classname
+        $serverName = Validator::validateClass($serverName, $this->config->get('namespaceName'));
+
+        // uppercase the name
+        $serverName = ucfirst($serverName);
+
+        // Create the class object
+        $comment = new PhpDocComment($this->description);
+        $this->class = new PhpClass($serverName, false, $this->config->get('soapServerClass'), $comment, false, true);
+
+        // Create the constructor
+        $comment = new PhpDocComment();
+        $comment->addParam(PhpDocElementFactory::getParam('array', 'options', 'A array of config values'));
+        $comment->addParam(PhpDocElementFactory::getParam('string', 'wsdl', 'The wsdl file to use'));
+
+        $source = '
+  foreach (' . $this->clientClassName . '::$classmap as $key => $value) {
+    if (!isset($options[\'classmap\'][$key])) {
+      $options[\'classmap\'][$key] = $value;
+    }
+  }' . PHP_EOL;
+        $source .= '  $options = array_merge(' . var_export($this->config->get('soapServerOptions'), true) . ', $options);' . PHP_EOL;
+        $source .= '  $this->computedOptions = $options;' . PHP_EOL;
+        $source .= '  if (!$wsdl) {' . PHP_EOL;
+        $source .= '    $wsdl = \'' . $this->config->get('inputFile') . '\';' . PHP_EOL;
+        $source .= '  }' . PHP_EOL;
+        $source .= '  parent::__construct($wsdl, $options);' . PHP_EOL;
+        $source .= '  $this->setObject($this);' . PHP_EOL;
+
+        $function = new PhpFunction('public', '__construct', 'array $options = array(), $wsdl = null', $source, $comment);
+
+        // Add the constructor
+        $this->class->addFunction($function);
+
+        // Generate the computed options
+        $name = 'computedOptions';
+        $comment = new PhpDocComment();
+        $comment->setVar(PhpDocElementFactory::getVar('array', $name, 'The computed options'));
+        $var = new PhpVariable('public', $name, '', $comment);
+
+        // Add the computed options variable
+        $this->class->addVariable($var);
```

Comment:
Why do we want to make the options available as a class property by default for the server? [We do not do this for the client](https://github.com/wsdl2phpgenerator/wsdl2phpgenerator/blob/master/src/Service.php#L140).